README.md
# Exile1201.info

other/changes.md
# Version 0.0.1

Added updates and changes to site.


js/home.js
$.holdReady(true);
var db = {};
var d0 = $.getJSON("json/home.json",function(data){
        db["homedata"] = data;
    });
$.when(d0).then(function(){
    $.holdReady(false);
});
    
    $(document).ready(function(){
		$('#to-tag').click(function(){      // When arrow is clicked
            $('body,html').animate({
                scrollTop : 0                       // Scroll to top of body
            }, 500);
        });

        $("#navitemLanguage").addClass('enable');
        
        var html = ""
        console.log(db.homedata)
        db.homedata.contents.forEach(eachcontent => {
            html += `
            <div class="featurebtn-container" onclick="window.location.href = '${eachcontent.link}'">
                <div class="featurebtn-open"><div>Open</div></div>
                <div class="featurebtn-button">
                    <div class="left"><img src="img/home/${eachcontent.image}.png">
                    <div class="onimage"><div class="titleonimage">${eachcontent.title}</div></div>
                    </div><div class="right"><div class="featurebtn-info">
                    <div class= "featurebtn-header"><h4>${eachcontent.title}</h4></div>
                    <div class="subfeature">Current Features :</div>
                    <ul class="home-feature-content" style="list-style-type: none; padding-left: 20px;">
            `
            eachcontent.content.forEach(element => {
                html+=`<li><i class="home-small">■</i> ${element}</li>`
            });
            html +=`
                        </ul>
                    </div>
                    </div>
                </div>
            </div>
            `
        });
        


        $('#home-content').html(html)
	});
	$.getScript("js/arrive.min.js", function(){
        $(document).arrive("#regionDropdown", function(){
            $("#navitemLanguage").addClass('enable');
        });
    });


js/lang.js
if(!localStorage.getItem('webLang')){
  localStorage.setItem("webLang", 'en');
  lang = "en";
  }
    switch (lang) {
        case "en":langText = 'English';break;
        case "cn":langText = 'Chinese';break;
        case "jp":langText = 'Japanese';break;
        case "kr":langText = 'Korean';break;
    }
// console.log(reg)
    let html = `
<div class="collapse navbar-collapse justify-content-between" id="navbarNav">
            <ul class="navbar-nav">
                    ${navDropdown.join("")}
                    
                    
                </ul>
<ul class="navbar-nav ml-auto">
                <li class="nav-item dropdown">
                    <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#ace-donatemodal"><img height='20px' style='border:0px;height:20px;margin-top:-5px' src='./img/items/GOLD.png'>  Donate</button>
                </li>
 <li class="nav-item dropdown" id="navitemLanguage">
                    <a class="nav-link dropdown-toggle" style="display:inline-flex;padding-left:25px"href="#" id="languageDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
                            <div class="ak-subtitle ak-disable" translate-id="language-2">Language</div>
                            <div class="ak-disable" id="display-lang">`+langText+`</div>
                    </a>
                    <div class="dropdown-menu" aria-labelledby="navbarDropdown">
                        <a class="dropdown-item lang unselectable" onclick="langDropdown($(this))" value="en">English</a>
                        <a class="dropdown-item lang unselectable" onclick="langDropdown($(this))" value="cn">Chinese</a>
                        <a class="dropdown-item lang unselectable" onclick="langDropdown($(this))" value="jp">Japanese</a>
                        <a class="dropdown-item lang unselectable" onclick="langDropdown($(this))" value="kr">Korean</a>
                    </div>
                </li>
            </ul>
</div>
`
    $('#aknav').html(html);
}


js/aknav.js


$(document).ready(function(){
    let dropdown = []
    dropdown.push({name:"Home",content:"index.html"})
    dropdown.push({
        name:"Calculator",
        content:[
            {
                href:"#",
                name:"test",
            },{
                href:"#",
                name:"Leveling"
            },{
                href:"#",
                name:"Material"
            },{
                href:"#",
                name:"Material Total Usage"
            }]
    })
    dropdown.push({
        name:"Information",
        content:[
        {
            href:"#",
            name:"Operator"
        },{
            href:"#",
            name:"Enemy"
        },{
            href:"#",
            name:"Infrastructure Skill"
        },{
            href:"#",
            name:"Comparing (WIP)"
        },{
            href:"#",
            name:"Stage (WIP)"
        },{
            href:"#",
            name:"Medal List (WIP)"
        },{
            href:"#",
            name:"Roguelike Relic List (WIP)"
        }]
    })
    dropdown.push({name:"Elite Materials",content:"#"})
    dropdown.push({name:"Guide",content:"#"})
    dropdown.push({name:"Mission Story",content:"#"})
    dropdown.push({
        name:"Extra",
        content:[
        {
            href:"#",
            name:"bilibili Linker"
        },
        {
            href:"#",
            name:"CG Gallery"
        }]
    })
      // console.log(window.location.href.split("/")[window.location.href.split("/").length-1])
    let currentHtml= window.location.href.split("/")[window.location.href.split("/").length-1];
    let navDropdown = []
    dropdown.forEach(drop => {

        let isCurrent = undefined
        let isGroup = undefined
        
        // console.log(typeof drop.content)
        if(typeof drop.content == "string"){
            if(currentHtml.includes(drop.content)) {
                isGroup = true;
            }
            navDropdown.push(`
            <li class="nav-item ${isGroup?"active":""}">
                <a class="nav-link " href="${drop.content}">${drop.name}</a>
            </li> 
            `)
        }else{
            drop.content.forEach(dropelement => {
                if(currentHtml.includes(dropelement.href)) {
                    isCurrent=dropelement.name;
                    isGroup = true;
                }
                
            })
            isCurrent = isCurrent?isCurrent:"Menu"
            navDropdown.push(`
            <li class="nav-item dropdown ${isCurrent!="menu"&&isGroup?"active":""}">
            <a class="nav-link dropdown-toggle" style="display:inline-flex;padding-left:25px;"href="#" id="regionDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" >
                    <div class="ak-subtitle ak-enable">${drop.name}</div>
            `)
            
            
            // console.log(isCurrent)
            navDropdown.push(`
                    <div class="ak-enable ">${isCurrent}</div>
                    </a>
                    <div class="dropdown-menu" aria-labelledby="navbarDropdown">
            `)
            // console.log(navDropdown)
            drop.content.forEach(dropelement => {
                navDropdown.push(`
                <a class="dropdown-item" href="${dropelement.href}" >${dropelement.name}</a>
                `)
            });
            navDropdown.push(`</div>`)
        }
    })
    let html = `
    <img src="./img/custom/texas.jpg" width="40" height="40" style="transform:scale(1.2,1.2)translate(-8px,1px)"class="d-inline-block align-top" alt="">
        <a class="navbar-brand" href="index.html" translate-id="topbar-1">Exile1201</a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav"
                aria-controls="navbarNav" aria-expanded="true" aria-label="展开">
            <span class="navbar-toggler-icon"></span>
        </button>
    
<div class="collapse navbar-collapse justify-content-between" id="navbarNav">
            <ul class="navbar-nav">

</ul>

`
    $('#aknav').html(html)


js/audio.js
var autoPlayAudios = function(){
	var autoPlayAudios = this;
	autoPlayAudios.audioDOMS = document.querySelectorAll('audio');
	
	//判断是否为微信浏览器
	autoPlayAudios.isWeiXin = function(){ 
		var ua = window.navigator.userAgent.toLowerCase(); 
		if(ua.match(/MicroMessenger/i) == 'micromessenger'){ 
			return true; 
		}else{ 
			return false; 
		} 
	} 
	
	//微信配置信息
	autoPlayAudios.autoPlayAudio = function() {
		wx.config({
            // 配置信息, 即使不正确也能使用 wx.ready
            debug: false,
            appId: '',
            timestamp: 1,
            nonceStr: '',
            signature: '',
            jsApiList: []
        });
        wx.ready(function() {
			autoPlayAudios.autoPlay();
        });
	}
	
	//自动播放
	autoPlayAudios.autoPlay  = function(){
		for(var i = 0 ; i < autoPlayAudios.audioDOMS.length ; i++){
			if(autoPlayAudios.audioDOMS[i].getAttribute('autoplay')!=null){
				autoPlayAudios.audioDOMS[i].play();
			}
		}
	}
	
	autoPlayAudios.play = function(){
		window.addEventListener('DOMContentLoaded',function(){
			//判断是否通过微信浏览器打开
//			alert(autoPlayAudios.isWeiXin());
			if(autoPlayAudios.isWeiXin()){
				var script = document.createElement('script');
				script.src = "http://res.wx.qq.com/open/js/jweixin-1.0.0.js";
				script.onload = function(){
					autoPlayAudios.autoPlayAudio();
				}
				document.body.appendChild(script);
			}else{
				//解决浏览器不会自动播放音乐
				var touchStartPlay = true;
				window.addEventListener('touchstart',function(){
					if(touchStartPlay){
						touchStartPlay = false;
						autoPlayAudios.autoPlay();
					}
				},false)
			}
		},false)
	}
}	


json/akhrhome.js
$.holdReady(true);
var db = {};
var d0 = $.getJSON("json/home.json",function(data){
        db["homedata"] = data;
    });
$.when(d0).then(function(){
    $.holdReady(false);
});
    
    $(document).ready(function(){
		$('#to-tag').click(function(){      // When arrow is clicked
            $('body,html').animate({
                scrollTop : 0                       // Scroll to top of body
            }, 500);
        });

	    $("#navitemRegion").addClass('ak-disable2');
        $("#navitemLanguage").addClass('ak-enable2');
        
        var html = ""
        console.log(db.homedata)
        db.homedata.contents.forEach(eachcontent => {
            html += `
            <div class="featurebtn-container" onclick="window.location.href = '${eachcontent.link}'">
                <div class="featurebtn-open"><div>Open</div></div>
                <div class="featurebtn-button">
                    <div class="left"><img src="img/ui/home/${eachcontent.image}.PNG">
                    <div class="onimage"><div class="titleonimage">${eachcontent.title}</div></div>
                    </div><div class="right"><div class="featurebtn-info">
                    <div class= "featurebtn-header"><h4>${eachcontent.title}</h4></div>
                    <div class="subfeature">Current Features :</div>
                    <ul class="home-feature-content" style="list-style-type: none; padding-left: 20px;">
            `
            eachcontent.content.forEach(element => {
                html+=`<li><i class="home-small">■</i> ${element}</li>`
            });
            html +=`
                        </ul>
                    </div>
                    </div>
                </div>
            </div>
            `
        });
        


        $('#home-content').html(html)
	});
	$.getScript("js/arrive.min.js", function(){
        $(document).arrive("#regionDropdown", function(){
            $("#navitemRegion").addClass('ak-disable2');
            $("#navitemLanguage").addClass('ak-enable2');
        });
    });


js/nav.js


$(document).ready(function(){
    let dropdown = []
    dropdown.push({name:"Home",content:"index.html"})
    dropdown.push({
        name:"About Me",
        content:[
            {
                href:"https://exile1201.github.io/blog/index.html",
                name:"About Me"
            },{
                href:"https://exile1201.github.io/blog/index.html",
                name:"Blogs",
            }]
    })
    dropdown.push({
        name:"Projects",
        content:[
        {
            href:"pixelgame.html",
            name:"Pixel Game"
        }]
    })
    dropdown.push({name:"Changes",content:"other/changes.html"})
    dropdown.push({name:"Guide",content:"#"})
    dropdown.push({name:"Mission Story",content:"#"})
    dropdown.push({
        name:"Extra",
        content:[
        {
            href:"#",
            name:"bilibili Linker"
        }]
    })
    // console.log(window.location.href.split("/")[window.location.href.split("/").length-1])
    let currentHtml= window.location.href.split("/")[window.location.href.split("/").length-1];
    let navDropdown = []
    dropdown.forEach(drop => {

        let isCurrent = undefined
        let isGroup = undefined
        
        // console.log(typeof drop.content)
        if(typeof drop.content == "string"){
            if(currentHtml.includes(drop.content)) {
                isGroup = true;
            }
            navDropdown.push(`
            <li class="nav-item ${isGroup?"active":""}">
                <a class="nav-link " href="${drop.content}">${drop.name}</a>
            </li> 
            `)
        }else{
            drop.content.forEach(dropelement => {
                if(currentHtml.includes(dropelement.href)) {
                    isCurrent=dropelement.name;
                    isGroup = true;
                }
                
            })
            isCurrent = isCurrent?isCurrent:"Menu"
            navDropdown.push(`
            <li class="nav-item dropdown ${isCurrent!="menu"&&isGroup?"active":""}">
            <a class="nav-link dropdown-toggle" style="display:inline-flex;padding-left:25px;"href="#" id="regionDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" >
                    <div class="subtitle disable">${drop.name}</div>
            `)
            
            
            // console.log(isCurrent)
            navDropdown.push(`
                    <div class="enable ">${isCurrent}</div>
                    </a>
                    <div class="dropdown-menu" aria-labelledby="navbarDropdown">
            `)
            // console.log(navDropdown)
            drop.content.forEach(dropelement => {
                navDropdown.push(`
                <a class="dropdown-item" href="${dropelement.href}" >${dropelement.name}</a>
                `)
            });
            navDropdown.push(`</div>`)
        }
    })
	

    // console.log(reg)
    let html = `
    <img src="./img/custom/texas.jpg" width="40" height="40" style="transform:scale(1.2,1.2)translate(-8px,1px)"class="d-inline-block align-top" alt="">
        <a class="navbar-brand" href="index.html" translate-id="topbar-1">Exile1201</a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav"
                aria-controls="navbarNav" aria-expanded="false" aria-label="展开">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse justify-content-between" id="navbarNav">
            <ul class="navbar-nav">
                    ${navDropdown.join("")}
                    
                    
                </ul>
            <ul class="navbar-nav ml-auto">
                <li class="nav-item dropdown">
                    <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#ace-donatemodal"><img height='20px' style='border:0px;height:20px;margin-top:-5px' src='./img/items/GOLD.png'>  Donate</button>
                </li>

                
                
            </ul>
        </div>


            </ul>
        </div>

    `
    $('#nav').html(html)

        var donate = `
    <div class="modal fade " id="ace-donatemodal">
        <div class="modal-dialog modal-dialog-centered modal" >
            <div class="modal-content" style="background:#444">
        
            <!-- Modal Header -->
            <!-- <div class="modal-header">
                <h4 class="modal-title"></h4>
                
            </div> -->
        
            <!-- Modal body -->
            <div class="modal-body acedonate-body">
            	<div style="text-align:left; margin: 5px;">
            		<h5>Support Me</h5>
            		<p>I make this website for myself in my free times, worth.</p>
            		<p>A dollar would help me live, plz...</p>
            		
            	</div>
            	<div class="row" style="padding: 10px; max-height: 160px; background-color: #333;">
            		<div class="col-md-6">
	            		<a class="btn btn-info" style="padding:15px; width: 100%; height: 100%" href='#' target='_blank'>
	            			<b style=" color: white;">Donate on Ko-fi</b>
	            			<div style="display:block; margin-top: 20px;">
	            				<img height="50" src="https://theme.zdassets.com/theme_assets/2141020/171bb773b32c4a72bcc2edfee4d01cbc00d8a004.png">
	            				<img height="30" style="filter: brightness(100)" src="https://pngimg.com/uploads/paypal/paypal_PNG4.png">
	            			</div>
	            		</a>
            		</div>
            		<div class="col-md-6">
	            		<a class="btn btn-info" style="padding:15px 10px 10px 10px; width: 100%;" href='#' target='_blank'>
	            			<b style=" color: white;">Donate on Trakteer</b>
	            			<div style="display:block;">
	            				<div style="background-color: #bf202f; display: inline-block; border-radius: 6px; padding: 5px 10px; vertical-align: middle;">
	            					<div style="background-image: url('https://trakteer.id/images/mix/navbar-logo-lite-white-beta.png'); background-size: cover; height: 43px; width: 28px; background-position: 0% 0%;"></div>
	            				</div>

	            				<div style="display: inline-block; padding: 5px 10px; position: relative; width: 100px; height: 90px; vertical-align: middle;">
	            					<div style="filter: brightness(100); height: 110px; width: 100px; left: 0px; top: -10px; position: absolute; background-position: 0% 0%; background-size: cover; background-image: url('./img/extra/gopay.png');"></div>
	            				</div>
	            			</div>
	            		</a>
            		</div>
            	</div>

            </div>
        
            <!-- Modal footer -->
            <div class="modal-footer" style="border-top: 1px solid #2f2f2f">
                <button type="button" class="btn btn-danger" data-dismiss="modal">Close</button>
            </div>
        
            </div>
        </div>
    </div>
    `
    $('#acedonate').html(donate)
});

function hideload(){
    $("#loadingOverlay").fadeOut(); 
}
